quantum/src/distance/distance_map.py

```diff
@@ -0,0 +1,68 @@
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+from logger import get_logger
+from protein import Protein
+from utils.qubit_utils import fix_qubits
+
+if TYPE_CHECKING:
+    from protein.bead import Bead
+
+logger = get_logger()
+
+
+class DistanceMap:
+    def __init__(self, protein: Protein):
+        self.protein = protein
+        self.distance_map = defaultdict(lambda: defaultdict(int))
+        self._distances_vector: list[defaultdict] = [
+            defaultdict(lambda: defaultdict(int)),
+            defaultdict(lambda: defaultdict(int)),
+            defaultdict(lambda: defaultdict(int)),
+            defaultdict(lambda: defaultdict(int)),
+        ]
```

Comment:
Maybe we could replace it with the code below and a constant `DIST_VECTOR_COUNT = 4`? It makes the code shorter and reduces repetition.

```suggestion
        self._distances_vector: list[defaultdict] = [
            defaultdict(lambda: defaultdict(int)) for _ in range(DIST_VECTOR_COUNT)
        ]
```